rpcs3/Emu/RSX/rsx_utils.cpp

```diff
@@ -135,60 +167,60 @@ namespace rsx
 
 	double avconf::get_aspect_ratio() const
 	{
-		video_aspect v_aspect;
 		switch (aspect)
 		{
-		case CELL_VIDEO_OUT_ASPECT_16_9: v_aspect = video_aspect::_16_9; break;
-		case CELL_VIDEO_OUT_ASPECT_4_3: v_aspect = video_aspect::_4_3; break;
-		case CELL_VIDEO_OUT_ASPECT_AUTO:
-		default: v_aspect = g_cfg.video.aspect_ratio; break;
-		}
-
-		double aspect_ratio;
-		switch (v_aspect)
-		{
-		case video_aspect::_4_3: aspect_ratio = 4.0 / 3.0; break;
-		case video_aspect::_16_9: aspect_ratio = 16.0 / 9.0; break;
+		case CELL_VIDEO_OUT_ASPECT_16_9: return 16. / 9.;
+		case CELL_VIDEO_OUT_ASPECT_4_3: return 4. / 3.;
+		default: fmt::throw_exception("Invalid aspect ratio %d", aspect);
```

Comment:
this change is pretty much fucking everything up.
The entire reason for this PR was the removed code above.
CELL_VIDEO_OUT_ASPECT_AUTO is a valid input here

Reply:
It's not. Auto is a throw-away parameter when the game is making requests and it is never returned. Confirmed with hwtests. It is there for making the API nicer to devs so that they don't need to check system details every time.

Reply:
The member "aspect" is CELL_VIDEO_OUT_ASPECT_AUTO by default until set by cellVideoOutConfigure.
The logic above handles this case by using the cfg param as also seen in cellVideoOutConfigure.
I can easily reproduce crashes with this code removed.

Reply:
If you have av resolution then aspect should be set to a real aspect. Auto should not be used that way, its the equivalent of passing a null parameter to mean "i don't care about the aspect just set the other stuff".
I'll add proper code to handle that.